flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/KvStateChangeLogger.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.state.changelog;
+
+import java.io.IOException;
+import java.util.Collection;
+
+interface KvStateChangeLogger<State, Namespace> extends StateChangeLogger<State, Namespace> {
+
+    void stateMerged(Namespace target, Collection<Namespace> sources) throws IOException;
```

Comment:
-> `mergeNamespaces` ?

Reply:
I think `stateMerged` is more consistent with parent interface methods (like `stateUpdated`) and better reflects that namespaces aren't actually merged here, but only logged.

Reply:
Things to sync up offline: I am confused why logging "name space merge" should be named as "state Merge"

Reply:
Renamed to `namespacesMerged`.